### ISSUE TITLE
Add support for ECDH-ES Key Management

### ISSUE THREAD

**Issue by nickjwallin** [airsidemobile/JOSESwift#171]
Hello, I would like to request that support for ECDH-ES key management be added :)

If it is not too much trouble, I am also curious as to how much effort it would be to implement this. I do not have much experience with cryptography, key agreement protocols, or encryption algorithms generally. Would it be hard to implement? Is implementation possible given CommonCrytpo's capabilities?

I did notice the existence of another repo that seems to do this: [JOSE-ECDH-ES-Swift](https://github.com/104corp/JOSE-ECDH-ES-Swift). Not sure if that implies that adding ECDH-ES here would be possible (or perhaps even simple).

Thank you!

**Comment by nickjwallin** [comment]
Hi @dnlggr, Happy New Year! 🎉 

Any word on if (or when) there are plans to add this functionality? :)

Thank you very much!

**Comment by daniel-mohemian** [comment]
Hi @nickjwallin,

Hope you had a great start into the new year!

Sorry for the late reply. Currently there are no plans for us to implement ECDH-ES key management by ourself in the near future. Regarding the other library you linked, it looks like they use [CrypoSwift](https://cryptoswift.io), although I'm not sure if they use it for the ECDH-ES key agreement part as I didn't look into it in that much detail.

TL;DR We gladly accept and help out with contributions utilizing native iOS cryptography primitives but do not plan to implement ECDH-ES by ourselves any time soon.
